### PR TITLE
MP Template Stability Update May 2024

### PR DESCRIPTION
**Pull request by GaianHelmers**
## What does this PR do?

Stability updates and fixes to the o3de-extras MPTemplate project.
- Removing corrupted icon data from Resources folder.
- Removing depreciated spawner code, updating lists and template file to reflect changes.
- Updating Demo level to support new simple spawning component.
- Tested working starting from scratch using Create Project window.
 - (I changed the Template folder body to dev branch o3de-extras template version)

Note: This update does not change the default Template button on the Project Manager. That is a separate issue here: https://github.com/o3de/o3de/issues/17989

## How was this PR tested?

1. Go to Create Project window in Project Manager.
2. Download the MPTemplate via the template buttons.
3. Navigate to your Template download destination. Delete the files in the MPTemplate folder.
4. Copy content from this versions MPTemplate into your local Template folder.
5. Create new project with that edited Template Folder.
6. The project should have a pink network preview image.
7. Build.
8. Build should succeed, may have a warning anyways.
9. Launch Editor even if the build has a warning icon.
10. Let the Asset Manager load.
11. Load Demo.level
12. Play from Editor after the boxes become visible due to asset processing completing.
13. You should be able to run around as BURT and shoot and crouch and things.

### CODE REVIEW

File: Templates/Multiplayer/template.json
Comment (AMZN-Gene) [inline Templates/Multiplayer/template.json:839]: +100 for red-code! How did you know to remove these IOSLauncher files?

**Review by AMZN-Gene — APPROVED** [review]

**Comment by AMZN-Gene** [comment]
Looks like we cannot merge this change while DCO is failing. When doing a git commit, you need to include -s (for sign), or manually add the sentence:
Signed-off-by: Username <email>

For example, when I do git commit -s, the commit message will add "Signed-off-by: AMZN-Gene <genewalt@amazon.com>"

This link should help you fix DCO
https://github.com/o3de/o3de-extras/pull/701/checks?check_run_id=25615613017

File: Templates/Multiplayer/template.json
Comment (GaianHelmers) [reply Templates/Multiplayer/template.json:839]: When I was pushing my project to git, git would fail to push with the clause that the files were corrupted. When I deleted, they had no lasting effect on the build or running of the project.

**Comment by GaianHelmers** [comment]
Fixed the DCO.

File: Templates/Multiplayer/template.json
Comment (GaianHelmers) [reply Templates/Multiplayer/template.json:839]: From the naming of the folders I intuit that it was lingering files from someone's attempt at building the project for release on a target platform. Which I don't believe should be a part of the Template. But I may be wrong.

**Comment by GaianHelmers** [comment]
Pushed one last fix to the level file. Hanging prefab references to a deleted spawner prefab were causing warnings.

**Comment by AMZN-Gene** [comment]
This PR says "continuous-integration/jenkins/pr-merge Expected — Waiting for status to be reported"
But I don't see PR #701 in Jenkins
https://jenkins.build.o3de.org/view/Projects/job/o3de-extras/view/change-requests/

**Comment by amzn-changml** [comment]
> This PR says "continuous-integration/jenkins/pr-merge Expected — Waiting for status to be reported" But I don't see PR #701 in Jenkins https://jenkins.build.o3de.org/view/Projects/job/o3de-extras/view/change-requests/

I’ve resolved the issue and running AR now

**Review by moudgils — APPROVED** [review]

**Comment by GaianHelmers** [comment]
@AMZN-Gene Got my second review! Are you able to merge it in?

File: Templates/Multiplayer/template.json
Comment (jhanca-robotecai) [inline Templates/Multiplayer/template.json:1074]: This file is still listed in [repo.json](https://github.com/GaianHelmers/o3de-extras/blob/303c9b5a0a8a2654d46254a577cf26a3a320a455/repo.json#L1488). I am aware this file is outdated, just saying...

Comment (AMZN-Gene) [reply Templates/Multiplayer/template.json:1074]: Good catch! Since the repo’s root repo.json won’t match, this will break the ability to download the MPTemplate directly inside of project manager (O3DE.exe). Because MPTemplate is already broken, I propose  merging this fix as a stepping stone for users who want to pull MPT via git, and then make another git hub issue for fixing the repo.json and release zip copy of mp template.

Comment (GaianHelmers) [reply Templates/Multiplayer/template.json:1074]: Ah darn. I manually edited the template file as I'm not actually familiar with how to re-templatize the project.

I agree with the stepping stone sentiment, as it's looking like there's a Gem configuration issue with the Template that needs fixing too. Discord discussion: https://discord.com/channels/805939474655346758/816043601754325023/1248334270313791600

**Comment by GaianHelmers** [comment]
@ShaunaGordon is looking into MP Template stability now. (For visibility)

**Comment by tkothadev** [comment]
What's the current status for MP Template Stability?

**Comment by GaianHelmers** [comment]
> What's the current status for MP Template Stability?

Unfortunately, we were not able to continue to diagnose the issues persisting beyond the work I had already done.

Currently, the project is able to be started and built from SDK. But due to the system being unable to "detect" the project Gem one is unable to use the dev branch editor.

The Project Gem is definitely in the right place and referenced correctly, however, I believe how it's configured is failing out and thus the engine is simply calling it "unfound".

Additionally, there's an issue of being unable to run the project in Play-Mode after doing _ANY_ edits due to the server that launches believing that the server version and the editor version are different, and thus it doesn't load in the MP scene at all.
This is the issue for it. Though the title could probably be changed: https://github.com/o3de/o3de/issues/17997

**Comment by tkothadev** [comment]
I see. It seems like the PR is stalled for the time being then, until those issues are resolved. Does this also mean the Issue https://github.com/o3de/o3de/issues/17989 is also stalled? 
If I understand correctly, this PR is a dependency for that issue.

**Comment by GaianHelmers** [comment]
@tkothadev Yes you are correct. There is a need for a clean up and diagnostics from someone who is more familiar with deploying templates and the MP functionality in general. I am a game dev, so I hit my depth a bit ago.

**Comment by GaianHelmers** [comment]
With this incoming fix: https://github.com/o3de/o3de-extras/pull/735
This PR can be closed.

**Comment by GaianHelmers** [comment]
Closed due to successful merger of MP Template Fix: https://github.com/o3de/o3de-extras/pull/735

**Pull request closed without merging**
